### ISSUE TITLE
Add decode() and decoding attribute to SVGImageElement

### ISSUE THREAD

**Issue by vmpstr** [w3c/svgwg#373]
The [HTMLImageElement](https://html.spec.whatwg.org/multipage/embedded-content.html#the-img-element) has recently been specced to include a [decode() API](https://html.spec.whatwg.org/multipage/embedded-content.html#dom-img-decode) and a [decoding attribute](https://html.spec.whatwg.org/multipage/embedded-content.html#attr-img-decoding).

I think that we should expand the [SVGImageElement spec](https://svgwg.org/svg2-draft/embedded.html#InterfaceSVGImageElement) to support similar functionality, since in spirit it should be able to support similar functionality as HTMLImageElements.

Thoughts?
/cc @smfr @annevk @chrishtr

**Comment by smfr** [comment]
feImage also has references to images.

**Comment by AmeliaBR** [comment]
The best way to do this would be to add the new methods to a shared interface which is then implemented by both element interfaces.

There is an [`HTMLOrSVGImageElement`](https://html.spec.whatwg.org/multipage/canvas.html#htmlorsvgimageelement) type defined for canvas, but it's currently not a proper partial interface or mixin.

**Comment by boggydigital** [comment]
Not blocking updated 2.0 CR publication - assigning 2.0 Recommendation milestone to clean this up before 2.0 REC

**Comment by dirkschulze** [comment]
I am not sure if it is still reasonable to bring this into 2.0 and we should defer it. Starting with the shared interface in the HTML spec as @AmeliaBR suggests sounds like a good idea.

**Comment by AmeliaBR** [comment]
Agreed. If there are no implementations, then it is not on track for SVG 2.0. Changing the Milestone accordingly.

That said, I am all for increased harmonization with HTML. I still think it would be best to approach this by working with HTML spec editors to define interface mixins so that new features can automatically be added to both types of image element simultaneously.

**Comment by dirkschulze** [comment]
@AmeliaBR I think that makes sense if implementations agree with the harmonization. We do have SVG 2.1 to keep on working on this.

**Comment by chrishtr** [comment]
This feature is already shipped in Chrome. We do need it to be specified somewhere.

Regarding "Work with HTML standards groups on better coordination for paired SVG/HTML elements", is there an existing precedent from another feature that this could be modeled on?

**Comment by annevk** [comment]
@chrishtr https://github.com/whatwg/html/commit/81e82437a7ef1ee37d4ce2b5a982bc297d5b58b1 is a somewhat recent commit that does something like what is wanted here. If you grep the whatwg/html commit log for SVG there's more further in the past.

**Comment by AmeliaBR** [comment]
Also relevant: the canvas APIs already define a [`HTMLOrSVGImageElement`](https://html.spec.whatwg.org/multipage/canvas.html#htmlorsvgimageelement) but only as a typedef. This would be about making that a mixin that defines the shared properties/methods.

(And similar for script and style.)
